src/main/java/com/bobocode/svydovets/beans/scanner/DefaultBeanScanner.java

```diff
@@ -0,0 +1,84 @@
+import com.bobocode.svydovets.beans.definition.BeanDefinition;
+import com.bobocode.svydovets.beans.definition.ConfigurationBeanDefinition;
+import com.bobocode.svydovets.exception.NoUniqueBeanDefinitionException;
+
+/**
+ * A bean definition scanner that detects bean candidates in configuration classes.
+ */
+public class DefaultBeanScanner implements BeanScanner {
+
+    /**
+     * This method scans package to find classes annotated with the @Configuration annotation
+     * and scans all methods annotated with the @Bean annotation within a provided configuration class.
+     * Returns bean definition for found methods and configuration classes annotated with appropriate annotations.
+     *
+     * @param packageName - package name that will be scanned
+     * @return - map of bean definitions
+     * @throws NoUniqueBeanDefinitionException in case of duplicating bean names
+     */
+    @Override
+    public Map<String, BeanDefinition> scan(String packageName) throws NoUniqueBeanDefinitionException {
+
+        var configurations = new Reflections(packageName)
+          .getTypesAnnotatedWith(Configuration.class)
+          .stream()
+          .map(this::getBeanDefinition)
+          .collect(toMap(BeanDefinition::getName, Function.identity()));
+
+        var beans = configurations.values().stream()
+          .flatMap(confClass -> Arrays.stream(confClass.getBeanClass().getMethods()))
+          .filter(method -> method.isAnnotationPresent(Bean.class))
+          .map(this::getConfigurationBeanDefinition)
+          .collect(toMap(BeanDefinition::getName, Function.identity()));
+
+        try {
+            return Stream.of(configurations, beans)
+              .flatMap(map -> map.entrySet().stream())
+              .collect(Collectors
+                .toMap(Map.Entry::getKey, Map.Entry::getValue));
+        } catch (IllegalStateException ex) {
+            throw new NoUniqueBeanDefinitionException(ex.getMessage());
+        }
+
+    }
+
+    /**
+     * Constructs a BeanDefinition instance based on configuration class parameter.
+     *
+     * @param confClass - class annotated with @Configuration annotation
+     * @return a new instance of a BeanDefinition
+     */
+    private BeanDefinition getBeanDefinition(Class<?> confClass) {
+        var confName = confClass.getAnnotation(Configuration.class).value().trim();
+        confName = confName.equals("") ? confClass.getSimpleName() : confName;
+        return new BeanDefinition(confName, confClass);
+    }
+
+    /**
+     * Constructs a ConfigurationBeanDefinition instance based on bean method parameter.
+     *
+     * @param beanMethod - a method annotated with @Bean annotation
+     * @return a new instance of a ConfigurationBeanDefinition
+     */
+    private ConfigurationBeanDefinition getConfigurationBeanDefinition(Method beanMethod) {
+        var beanName = beanMethod.getAnnotation(Bean.class).value().trim();
+        var beanClass = beanMethod.getReturnType();
```

Comment:
if a user makes 'void' method and marks it as @Bean, then should be thrown an exception

Reply:
fixed